### PR TITLE
check and warn if uid exists before setting

### PR DESCRIPTION
**Pull request by sunew**

### CODE REVIEW

File: src/plone/app/transmogrifier/uidupdater.py [hunk 1]
```diff
@@ -59,6 +64,12 @@ def __iter__(self):
                         obj._setUID(uid)
 
             if IAttributeUUID.providedBy(obj):
+                current_uid = IUUID(obj.aq_base, None)
+                existing_uid = obj.portal_catalog({'UID': uid})
+                if bool(existing_uid):
+                    logger.debug("Existing UID {} on {}, this item: {}".format(
```
Comment (hvelarde) [inline src/plone/app/transmogrifier/uidupdater.py]: this is not a warning, you should use `logger.debug()` instead; also, don't use `{}` as we will lose compatibility with Python < 2.7, IIRC.

**Comment by sunew** [comment]
@hvelarde fixed

File: src/plone/app/transmogrifier/uidupdater.py [hunk 2]
```diff
@@ -59,6 +63,11 @@ def __iter__(self):
                         obj._setUID(uid)
 
             if IAttributeUUID.providedBy(obj):
+                existing_uid = obj.portal_catalog({'UID': uid})
+                if bool(existing_uid):
+                    logger.debug('When updating UID for %s, another item with the same UID %s was found here: %s' % (
```
Comment (hvelarde) [inline src/plone/app/transmogrifier/uidupdater.py]: sorry, I meant `logger.warn()` instead.

also, I don't know if using acquisition is better/faster than fetching the `portal_catalog` the standard way. do you mind to comment on this?

File: src/plone/app/transmogrifier/uidupdater.py [hunk 3]
```diff
@@ -9,6 +9,10 @@
 from plone.uuid.interfaces import IAttributeUUID
 from plone.uuid.interfaces import IMutableUUID
 from zope.interface import classProvides, implements
+import logging
+
+
+logger = logging.getLogger("plone.app.transmogrifier uidupdater")
```
Comment (hvelarde) [inline src/plone/app/transmogrifier/uidupdater.py]: any reason not to use plain `plone.app.transmogrifier` logger?

Comment (sunew) [reply src/plone/app/transmogrifier/uidupdater.py]: Yes, this is the same pattern the other blueprints are using - so its possible to see from the logger name what blueprint is used (see collective/transmogrifier/sections/constructor or plone/app/transmogrifier/reindexobject)

File: src/plone/app/transmogrifier/uidupdater.py (hunk 2 above)
Comment (sunew) [reply src/plone/app/transmogrifier/uidupdater.py]: catalog: you are right - I have changed it so its similar to plone/app/transmogrifier/reindexobject.py

**Comment by sunew** [comment]
To facilitate multiple runs of the same pipeline, I added a check to not warn if it is the same object.

**Comment by ale-rt** [comment]
I am not sure why this is needed.
AFAIK you are already warned if you try to set the same UID to different objects.

**Pull request closed without merging**